server/image_pull.go

```diff
@@ -18,24 +18,17 @@ import (
 
 var localRegistryPrefix = libpodImage.DefaultLocalRegistry + "/"
 
-// PullImage pulls a image with authentication config.
-func (s *Server) PullImage(ctx context.Context, req *pb.PullImageRequest) (resp *pb.PullImageResponse, err error) {
-	// TODO: what else do we need here? (Signatures when the story isn't just pulling from docker://)
-	image := ""
-	img := req.GetImage()
-	if img != nil {
-		image = img.Image
-	}
-
+func (s *Server) pullImage(ctx context.Context, req *pb.PullImageRequest, image string) (string, error) {
```

Comment:
I generally prefer the called function to come after the calling function. That way it can be read top to bottom and go down the stack. Can you switch the location with PullImage?

Reply:
Also I think a comment explaining the reasoning behind having a public PullImage and private pullImage split up would be helpful

Reply:
Sounds good! I moved it down and made sure to add more verbose comments all around.